### PR TITLE
1117 - add reward pool fetching utils

### PR DESCRIPTION
**Pull request by Preston-Harrison**
linked to https://github.com/handle-fi/handle-react/issues/1117

### CODE REVIEW

File: src/utils/reward-utils.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                marginAndLiquidation
+                swap
+                mint
+                burn
+            }
+            hlpStats(first:1, orderBy:timestamp, orderDirection:desc) {
+                aumInUsdg
+            }
+        }
+    `;
```
Comment (rellfy) [inline src/utils/reward-utils.ts:48]: Could you make a getter for this query outside of the function?

File: src/utils/reward-utils.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  burn: string;
+};
+type GraphResponse = {
+  feeStats: FeeStats[];
+  hlpStats: {
+    aumInUsdg: string;
+  }[];
+};
+
+export const fetchFeeAPR = async (
```
Comment (rellfy) [inline src/utils/reward-utils.ts:29]: camelCase for function names please
```suggestion
export const fetchFeeApr = async (
```

File: src/utils/reward-utils.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import { GovernanceLock__factory, RewardPool__factory } from "../contracts";
+import { Network } from "../types/network";
+
+export const calculateUserAPR = (
+  userBoost: number,
+  averagePoolBoost: number,
+  poolAPR: number
+): number => {
+  return (poolAPR * userBoost) / averagePoolBoost;
+};
```
Comment (rellfy) [inline src/utils/reward-utils.ts:14]: :D
```suggestion
): number => (poolAPR * userBoost) / averagePoolBoost;
```

File: src/utils/reward-utils.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
+import { BigNumber, BigNumberish, ethers } from "ethers";
+import request, { gql } from "graphql-request";
+import config from "../config";
+import { BASIS_POINTS_DIVISOR, PRICE_DECIMALS } from "../config/hlp";
+import { GovernanceLock__factory, RewardPool__factory } from "../contracts";
+import { Network } from "../types/network";
+
+export const calculateUserAPR = (
```
Comment (rellfy) [inline src/utils/reward-utils.ts:8]:
```suggestion
export const calculateUserApr = (
```

**Review by nabsta-eth — APPROVED** [review]

File: src/utils/reward.ts [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    .mul(15)
+    .div(totalSupply)
+    .div(10);
+  const boost = deposit.amount.add(boostDepositAmount);
+  let boostRatio = boost.mul(BASIS_POINTS_DIVISOR).div(deposit.amount);
+  if (boostRatio.gt(2.5 * BASIS_POINTS_DIVISOR)) {
+    boostRatio = BigNumber.from(2.5 * BASIS_POINTS_DIVISOR);
+  }
+
+  return 1 + +boostRatio / BASIS_POINTS_DIVISOR;
```
Comment (rellfy) [inline src/utils/reward.ts:122]: Boost ratio is up to 2.5, not 1.5, so we don't need to add 1 here
```suggestion
  return +boostRatio / BASIS_POINTS_DIVISOR;
```

**Review by nabsta-eth — APPROVED** [review]

**Review by rellfy — APPROVED** [review]

**Pull request merged**